### ISSUE TITLE
Release v0.2.0

### ISSUE THREAD

**Issue by danielvegamyhre** [kubernetes-sigs/jobset#207]
## Release Checklist

- [x] All [OWNERS](https://github.com/kubernetes-sigs/jobset/blob/main/OWNERS) must LGTM the release proposal
- [x] Verify that the changelog in this issue is up-to-date
- [x] For major or minor releases (v$MAJ.$MIN.0), create a new release branch.
  - [x] an OWNER creates a vanilla release branch with
        `git branch release-$MAJ.$MIN main`
  - [x] An OWNER pushes the new release branch with
        `git push release-$MAJ.$MIN`
- [x] Update things like README, deployment templates, docs, configuration, test/e2e flags.
      Submit a PR against the release branch: 
- [x] An OWNER [prepares a draft release](https://github.com/kubernetes-sigs/jobset/releases)
  - [x] Write the change log into the draft release.
  - [x] Run
      `make artifacts IMAGE_REGISTRY=registry.k8s.io/jobset GIT_TAG=$VERSION`
      to generate the artifacts and upload the files in the `artifacts` folder
      to the draft release.
- [x] An OWNER creates a signed tag running
     `git tag -s $VERSION`
      and inserts the changelog into the tag description.
      To perform this step, you need [a PGP key registered on github](https://docs.github.com/en/authentication/managing-commit-signature-verification/checking-for-existing-gpg-keys).
- [x] An OWNER pushes the tag with
      `git push $VERSION`
  - Triggers prow to build and publish a staging container image
      `gcr.io/k8s-staging-jobset/jobset:$VERSION`
- [x] Submit a PR against [k8s.io](https://github.com/kubernetes/k8s.io), 
      updating `k8s.gcr.io/images/k8s-staging-jobset/images.yaml` to
      [promote the container images](https://github.com/kubernetes/k8s.io/tree/main/k8s.gcr.io#image-promoter)
      to production: 
- [x] Wait for the PR to be merged and verify that the image `registry.k8s.io/jobset/jobset:$VERSION` is available.
- [x] Publish the draft release prepared at the [Github releases page](https://github.com/kubernetes-sigs/jobset/releases).
- [x] Add a link to the tagged release in this issue: 
- [x] Send an announcement email to `sig-apps@kubernetes.io`, `sig-scheduling@kubernetes.io` and `wg-batch@kubernetes.io` with the subject `[ANNOUNCE] JobSet $VERSION is released`
- [x] Add a link to the release announcement in this issue: 
- [x] For a major or minor release, update `README.md` and `docs/setup/install.md`
      in `main` branch: 
- [x] For a major or minor release, create an unannotated _devel_ tag in the
      `main` branch, on the first commit that gets merged after the release
       branch has been created (presumably the README update commit above), and, push the tag:
      `DEVEL=v0.$(($MAJ+1)).0-devel; git tag $DEVEL main && git push $DEVEL`
      This ensures that the devel builds on the `main` branch will have a meaningful version number.
- [x] Close this issue

## Changelog
- Added tracking for the status of ReplicatedJobs (number of active/succeeded/failed jobs for each replicated job) to support Kueue integration.
- NodeAffinity is now mutable for suspended JobSets to support Kueue integration. When the JobSet is unsuspended, any ReplicatedJob with a modified NodeAffinity will result in the child jobs having their NodeAffinities updated to match.
- Added Go and Python client libraries for interacting with JobSet API
- JobSets now have a configurable SuccessPolicy. The policy targets a subset of ReplicatedJobs, and can wait for any or all jobs within each ReplicatedJob to complete before marking the JobSet complete.
- 1 headless service per JobSet is created, instead of 1 headless service per ReplicatedJob.
- User can specify the network subdomain, otherwise it will be defaulted to the JobSet name.
- Bug fix ensuring we only count jobs defined in the JobSet spec when determining if the JobSet is completed or not.
- Testgrid improvements (each test name shows up with success/failure now).

**Comment by danielvegamyhre** [comment]
Verified v0.2.0 image is in prod:

```
$ docker pull registry.k8s.io/jobset/jobset:v0.2.0
v0.2.0: Pulling from jobset/jobset
a7ca0d9ba68f: Already exists 
fe5ca62666f0: Already exists 
b02a7525f878: Already exists 
fcb6f6d2c998: Already exists 
e8c73c638ae9: Already exists 
1e3d9b7d1452: Already exists 
4aa0ea1413d3: Already exists 
7c881f9ab25e: Already exists 
5627a970d25e: Already exists 
cc1189d6bb47: Pull complete 
Digest: sha256:355659bfd56d04745c3fdb0c029369dbdfa05433717157ee5b8f5ea09046bfb9
Status: Downloaded newer image for registry.k8s.io/jobset/jobset:v0.2.0
registry.k8s.io/jobset/jobset:v0.2.0
```

**Comment by danielvegamyhre** [comment]
Published release v0.2.0: https://github.com/kubernetes-sigs/jobset/releases/tag/v0.2.0

**Comment by danielvegamyhre** [comment]
Link to announcement email: https://groups.google.com/a/kubernetes.io/g/wg-batch/c/aELRWx5LJ80/m/JNi0EgYFBAAJ
